modules/swagger-codegen/src/main/resources/TypeScript-Angular/api.mustache

```diff
@@ -0,0 +1,68 @@
+/// <reference path="api.d.ts" />
+
+/* tslint:disable:no-unused-variable member-ordering */
+
+{{#operations}}
+module {{package}} {
+    'use strict';
+
+    {{#description}}
+    /**
+    * {{&description}}
+    */
+    {{/description}}
+    export class {{classname}} {
+        private basePath = '{{basePath}}';
+
+        static $inject: string[] = ['$http'];
+
+        constructor(private $http: ng.IHttpService, basePath?: string) {
+            if (basePath) {
+                this.basePath = basePath;
+            }
+        }
+        {{#operation}}
+        public {{nickname}} ({{#allParams}}{{paramName}}: {{{dataType}}}, {{/allParams}} extraHttpRequestParams?: any ) : ng.IHttpPromise<{{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}{}{{/returnType}}> {
+            var path = this.basePath + '{{path}}';
+            {{#pathParams}}
+            path = path.replace('{' + '{{paramName}}' + '}', String({{paramName}}));
+            {{/pathParams}}
+            var queryParameters: any = {};
+            var headers: any = {};
+            {{#requiredParamCount}}
```

Comment:
I'm afraid this wont' work. If you look at the sample generated, the getPetById method does not verify the `id` as a required field.

I worked around the issue by using {{#allParams}}{{required}} ... {{/required}}{{/allParams}}

Please refer to https://github.com/swagger-api/swagger-codegen/blob/develop_2.0/modules/swagger-codegen/src/main/resources/php/api.mustache#L67 for more information and let me know if you need any help.